src/directions/helpers.ts

```diff
@@ -202,3 +203,200 @@ export function getReverseGeocodedAddresses(
+ *
+ * @param request Google.maps.DirectionsRequest or google.maps.DistanceMatrixRequest while making route or
+ *   getDistanceMatrix call
+ * @param response - OriginReponse or OriginsResponse containing location information
+ * @returns {undefined} Longitude, latitude array or null if coordinates cannot be extracted
+ */
+// keep any for now, need to fix _convertAmazonResponseToGoogleResponse types first
+export function extractCoordinates(
+  request: google.maps.DirectionsRequest | google.maps.DistanceMatrixRequest,
+  response: any,
+): number[] | null {
+  if (isDirectionsRequest(request)) {
+    return extractDirectionsOrigin(response);
+  }
+
+  return extractDistanceMatrixOrigin(response);
+}
+
+/**
+ * Checks if the given request is a DirectionsRequest.
+ *
+ * This type guard function determines if the input object has both 'origin' and 'destination' properties, which are
+ * characteristic of a Google Maps DirectionsRequest.
+ *
+ * @param request - The object to be checked
+ * @returns True if the request is a DirectionsRequest, false otherwise
+ */
+// keep any for now, need to fix _convertAmazonResponseToGoogleResponse types first
+function isDirectionsRequest(request: any): request is google.maps.DirectionsRequest {
+  return "origin" in request && "destination" in request;
+}
+
+/**
+ * Extracts coordinates from OriginResponse response
+ *
+ * @param response - OriginResponse containing location information
+ * @returns {undefined} Longitude, latitude array or null if coordinates cannot be extracted
+ */
+function extractDirectionsOrigin(response): number[] | null {
+  const lat = response?.locationLatLng?.lat?.();
+  const lng = response?.locationLatLng?.lng?.();
+
+  return typeof lat === "number" && typeof lng === "number" ? [lat, lng] : null;
+}
+
+/**
+ * Extracts coordinates from OriginsResponse response
+ *
+ * @param response - OriginsResponse containing location information
+ * @returns {undefined} Longitude, latitude array or null if coordinates cannot be extracted
+ */
+function extractDistanceMatrixOrigin(response): number[] | null {
+  const lat = response?.[0]?.locationLatLng?.lat?.();
+  const lng = response?.[0]?.locationLatLng?.lng?.();
+
+  return typeof lat === "number" && typeof lng === "number" ? [lat, lng] : null;
+}
+
+/**
+ * Formats a distance value based on the specified unit system (metric or imperial).
+ *
+ * Metric formatting rules:
+ *
+ * 1. < 1 km: Format in meters, rounded to nearest meter ("750 m")
+ * 2. 1 km to 999 km: Format in km with one decimal place ("12.5 km", "542.0 km")
+ * 3. > = 1000 km: Format in km with no decimal places and thousands separator ("1,234 km")
+ *
+ * Imperial formatting rules:
+ *
+ * 1. < 0.1 miles: Format in feet, rounded to nearest foot ("528 ft")
+ * 2. 0.1 miles to 999 miles: Format in miles with one decimal place ("0.5 mi", "12.0 mi", "542.0 mi")
+ * 3. > = 1000 miles: Format in miles with no decimal places and thousands separator ("1,234 mi")
```

Comment:
Great docs here! Did this come from Google's documentation? If so we should link to it.